wherehows-frontend/app/security/AuthenticationManager.java

```diff
@@ -13,174 +13,58 @@
-    if (StringUtils.isBlank(userName)) {
-      throw new IllegalArgumentException("Username and password can not be blank.");
-    }
-
-    if (attributeNames.length == 0) {
-      return Collections.emptyMap();
-    }
-
-    SearchControls searchControls = new SearchControls();
-    searchControls.setSearchScope(SearchControls.SUBTREE_SCOPE);
-    searchControls.setReturningAttributes(attributeNames);
-
-    NamingEnumeration<? extends SearchResult> searchResult =
-        ctx.search(searchBase, LDAP_USER_ATTR_FILTER_EXPRESSION, new Object[]{userName}, searchControls);
-
-    Map<String, String> result = new HashMap<>();
-    if (searchResult.hasMore()) {
-      NamingEnumeration<? extends Attribute> attributes = searchResult.next().getAttributes().getAll();
-
-      while (attributes.hasMore()) {
-        Attribute attr = attributes.next();
-        String attrId = attr.getID();
-        String attrValue = (String) attr.get();
-
-        result.put(attrId, attrValue);
-      }
+  private static class WHZCallbackHandler implements CallbackHandler {
+    private String password = null;
+    private String username = null;
+    private WHZCallbackHandler(String username, String password) {
+      this.username = username;
+      this.password = password;
     }
-    return result;
-  }
-
-  private static User getAttributes(DirContext ctx, String searchBase, String userName, String principalDomain)
-      throws NamingException {
-    Map<String, String> userDetailMap =
-        getUserAttributes(ctx, searchBase, userName, LDAP_DISPLAY_NAME_KEY, LDAP_MAIL_KEY, LDAP_DEPARTMENT_NUMBER_KEY);
-    Logger.info("userDetailMap: " + userDetailMap);
 
-    String displayName = userDetailMap.get(LDAP_DISPLAY_NAME_KEY);
-    String[] displayNameTokens = displayName.trim().replaceAll(" +", " ").split(" ");
-    String firstName = displayNameTokens[0];
-    String lastName = displayNameTokens[1];
-    String email = userDetailMap.get(LDAP_MAIL_KEY);
-    String department = userDetailMap.get(LDAP_DEPARTMENT_NUMBER_KEY);
-    int departmentNum = 0;
-    if (StringUtils.isNotBlank(department)) {
-      try {
-        departmentNum = Integer.parseInt(department);
-      } catch (NumberFormatException e) {
-        Logger.error("Convert department number failed. Error message: " + e.getMessage());
-        departmentNum = 0;
+    @Override
+    public void handle(Callback[] callbacks)
+        throws UnsupportedCallbackException {
+      NameCallback nc = null;
+      PasswordCallback pc = null;
+      for (Callback callback : callbacks) {
+        if (callback instanceof NameCallback) {
+          nc = (NameCallback) callback;
+          nc.setName(this.username);
+        } else if (callback instanceof PasswordCallback) {
+          pc = (PasswordCallback) callback;
+          pc.setPassword(this.password.toCharArray());
+        } else {
+          throw new UnsupportedCallbackException(callback, "The submitted Callback is unsupported");
```

Comment:
Maybe just log warning instead of throwing exception in case some LDAP implementation oddity.